pkg/controllers/customroute/custom_route_controller.go

```diff
@@ -0,0 +1,509 @@
+				Namespace: "openshift-authentication"},
+		},
+		Conditions: c.getConditions(customRouteErrors, existingRoute, ingress),
+	})
+	return err
+}
+
+func (c *customRouteController) getConditions(errs []error, route *routev1.Route, ingress *configv1.Ingress) []metav1.Condition {
+	now := metav1.Now()
+	if len(errs) != 0 {
+		return []metav1.Condition{
+			{
+				LastTransitionTime: now,
+				Type:               "Degraded",
+				Status:             metav1.ConditionTrue,
+				Reason:             "CustomRouteError",
+				Message:            fmt.Sprintf("Error Configuring custom route: %v", errs),
+			},
+			{
+				LastTransitionTime: now,
+				Type:               "Progressing",
+				Status:             metav1.ConditionFalse,
+				Reason:             "CustomRouteError",
+				Message:            fmt.Sprintf("Error Configuring custom route: %v", errs),
+			},
+		}
+	}
+
+	if _, _, err := routeapihelpers.IngressURI(route, route.Spec.Host); err != nil {
+		return []metav1.Condition{
+			{
+				LastTransitionTime: now,
+				Type:               "Degraded",
+				Status:             metav1.ConditionTrue,
+				Reason:             "RouteNotAdmitted",
+				Message:            fmt.Sprintf("Route not admitted: %v", err),
+			},
+			{
+				LastTransitionTime: now,
+				Type:               "Progressing",
+				Status:             metav1.ConditionFalse,
+				Reason:             "RouteNotAdmitted",
+				Message:            fmt.Sprintf("Route not admitted: %v", err),
+			},
+		}
+	}
+
+	if err := c.checkRouteAvailable(route.Spec.Host, ingress); err != nil {
+		return []metav1.Condition{
+			{
+				LastTransitionTime: now,
+				Type:               "Degraded",
+				Status:             metav1.ConditionFalse,
+				Reason:             "ErrorReachingOutToService",
+				Message:            fmt.Sprintf("unexpected error at %s: %v", route.Spec.Host, err),
+			},
+			{
+				LastTransitionTime: now,
+				Type:               "Progressing",
+				Status:             metav1.ConditionTrue,
+				Reason:             "ErrorReachingOutToService",
+				Message:            fmt.Sprintf("unexpected error at %s: %v", route.Spec.Host, err),
+			},
+		}
+	}
+
+	return []metav1.Condition{
+		{
+			LastTransitionTime: now,
+			Type:               "Degraded",
+			Status:             metav1.ConditionFalse,
+			Reason:             "RouteIsAvailable",
+			Message:            "Route successfully configured",
+		},
+		{
+			LastTransitionTime: now,
+			Type:               "Progressing",
+			Status:             metav1.ConditionFalse,
+			Reason:             "RouteIsAvailable",
+			Message:            "Route successfully configured",
+		},
+	}
+}
+
+func (c *customRouteController) checkRouteAvailable(host string, ingress *configv1.Ingress) error {
+	url := "https://" + host
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return err
+	}
+
+	reqCtx, cancel := context.WithTimeout(context.TODO(), 10*time.Second) // avoid waiting forever
+	defer cancel()
+	req.WithContext(reqCtx)
```

Comment:
this does nothing, look at `http.NewRequestWithContext` (I learned this yesterday)

Reply:
I'll use the `http.NewRequestWithContext` as suggested, thanks!

Reply:
[This bit](https://golang.org/pkg/net/http/#Request.WithContext) of documentation for `WithContext` was helpful as well.